dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -150,6 +150,7 @@
         this.id = archiveID;
         this.eeModuleDescptorId = eeModuleDescptorId == null ? archiveID : eeModuleDescptorId;
         this.archive = archive;
+        this.classloader = archive.getClassLoader();
```

Comment:
Notice that there is a parameter on the constructor here which is also called allClasses ... now that they are both Set<String> one could mask the other and so should be renamed.